### PR TITLE
LPS-185768 Remove AMImageSizeProvider

### PR DESCRIPTION
**Pull request by julschong**
resubmitted from https://github.com/liferay-core-infra/liferay-portal/pull/4898
- Added upgrade process in -service module

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:lima

**Comment by julschong** [comment]
ci:test:upgrade

**Comment by Kvale1733** [comment]
@julschong No regressions. Just need to update configuration properties

com.liferay.portal.tools.ConfigurationEnvBuilderTest
```
    [junit] Testcase: testBuildContent(com.liferay.portal.tools.ConfigurationEnvBuilderTest):	FAILED
    [junit] Run "ant update-portal-osgi-configuration-properties". Leftover: [configuration.override.com.liferay.adaptive.media.image.internal.configuration.AMImageConfiguration_imageMaxSize=]
    [junit] junit.framework.AssertionFailedError: Run "ant update-portal-osgi-configuration-properties". Leftover: [configuration.override.com.liferay.adaptive.media.image.internal.configuration.AMImageConfiguration_imageMaxSize=]
    [junit] 	at com.liferay.portal.tools.ConfigurationEnvBuilderTest.testBuildContent(ConfigurationEnvBuilderTest.java:107)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:99)
    [junit] 	at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:59)
```
[Log](https://testray.liferay.com/reports/production/logs/2023-06/test-1-26/test-portal-acceptance-pullrequest(master)/10204/unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by julschong** [comment]
added ant update-portal-osgi-configuration-properties commit

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by julschong** [comment]
@tinatian changed upgradeProcess to not deal with empty config file

File: modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/v2_3_0/AMImageConfigurationUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+				"(", Constants.SERVICE_PID,
+				"=com.liferay.adaptive.media.image.internal.configuration.",
+				"AMImageConfiguration)"));
+
+		if (configurations == null) {
+			return;
+		}
+
+		Configuration configuration = configurations[0];
+
```
Comment (julschong) [inline modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/v2_3_0/AMImageConfigurationUpgradeProcess.java:53]: changed here

**Comment by julschong** [comment]
ci:test:sf

File: modules/apps/adaptive-media/adaptive-media-image-service/bnd.bnd [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Adaptive Media Image Service
 Bundle-SymbolicName: com.liferay.adaptive.media.image.service
 Bundle-Version: 5.0.35
-Liferay-Require-SchemaVersion: 2.2.0
+Liferay-Require-SchemaVersion: 2.3.0
```
Comment (tinatian) [inline modules/apps/adaptive-media/adaptive-media-image-service/bnd.bnd:4]: @julschong I do not think you need to do a minor version change, a micro version change should be enough as there is actually no database table change at all.

Comment (julschong) [reply modules/apps/adaptive-media/adaptive-media-image-service/bnd.bnd:4]: Ok, let me change it

File: modules/apps/adaptive-media/adaptive-media-image-service/bnd.bnd [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Adaptive Media Image Service
 Bundle-SymbolicName: com.liferay.adaptive.media.image.service
 Bundle-Version: 5.0.35
-Liferay-Require-SchemaVersion: 2.2.0
+Liferay-Require-SchemaVersion: 2.2.1
```
Comment (julschong) [inline modules/apps/adaptive-media/adaptive-media-image-service/bnd.bnd:4]: changed from 2.3.0 to 2.2.1

File: modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/registry/AMImageServiceUpgradeStepRegistrator.java [hunk 4]
```diff
@@ -49,6 +52,13 @@ protected String[] getTableNames() {
 
 		registry.register(
 			"2.1.0", "2.2.0", new CTModelUpgradeProcess("AMImageEntry"));
+
+		registry.register(
+			"2.2.0", "2.2.1",
```
Comment (julschong) [inline modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/registry/AMImageServiceUpgradeStepRegistrator.java:57]: changed from 2.3.0 to 2.2.1

File: modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/v2_2_1/AMImageConfigurationUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.adaptive.media.image.internal.upgrade.v2_2_1;
```
Comment (julschong) [inline modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/v2_2_1/AMImageConfigurationUpgradeProcess.java:15]: changed from 2.3.0 to 2.2.1

File: modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/registry/AMImageServiceUpgradeStepRegistrator.java [hunk 6]
```diff
@@ -15,12 +15,15 @@
 package com.liferay.adaptive.media.image.internal.upgrade.registry;
 
 import com.liferay.adaptive.media.image.internal.upgrade.v2_0_0.util.AMImageEntryTable;
+import com.liferay.adaptive.media.image.internal.upgrade.v2_2_1.AMImageConfigurationUpgradeProcess;
```
Comment (julschong) [inline modules/apps/adaptive-media/adaptive-media-image-service/src/main/java/com/liferay/adaptive/media/image/internal/upgrade/registry/AMImageServiceUpgradeStepRegistrator.java:18]: changed from 2.3.0 to 2.2.1

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
rebased to resolve conflict in breaking change

**Comment by tinatian** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/137692

**Pull request closed without merging**